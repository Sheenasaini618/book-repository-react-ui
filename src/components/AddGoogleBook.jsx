import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import useMediaQuery from '@mui/material/useMediaQuery';
import { useTheme } from '@mui/material/styles';
import Box from '@mui/material/Box';
import Fab from '@mui/material/Fab';
import AddIcon from '@mui/icons-material/Add';
import { addGoogleBook } from '../util/api/bookApi';

export default function AddGoogleBook(props) {

  var bookData = props.details

  const [open, setOpen] = React.useState(false);
  const theme = useTheme();
  // const fullScreen = useMediaQuery(theme.breakpoints.down('md'));

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleGoogleBookAdd = () => {
    addGoogleBook(bookData).then((res) => {
      console.log(res);
      handleClose();
      window.location.reload(false);
    });
  };

  return (
    <div>
      <Box sx={{ '& > :not(style)': { m: 1 } }}>
        <Fab size="small" color="primary" aria-label="add" onClick={handleClickOpen}>
          <AddIcon />
        </Fab>
        <Dialog
          open={open}
          onClose={handleClose}
          aria-labelledby="responsive-dialog-title"
        >
          <DialogTitle className='dialogTitle' id="responsive-dialog-title">
            {"Are you sure you want to add the book?"}
          </DialogTitle>
          <DialogActions>
            <Button className = 'closeButton' onClick={handleClose}>
              No
            </Button>
            <Button className = 'approveButton' onClick={handleGoogleBookAdd} autoFocus>
              Yes
            </Button>
          </DialogActions>
        </Dialog>
      </Box>
    </div>
  );
}