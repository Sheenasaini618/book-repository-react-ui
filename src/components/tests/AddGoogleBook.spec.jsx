import React from 'react'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import AddGoogleBook from '../AddGoogleBook';

const mySpecialWindowFunction = () => {
    return /msie [6-9]\b/.test(window.navigator.userAgent.toLowerCase());
};


configure({ adapter: new Adapter() });

global.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
        ok: true,
        status: 200,
    }),
);

describe('<AddGoogleBook />', () => {

    let props;
    let component;
    if (typeof window === 'undefined') {
        global.window = {}
    }

    beforeEach(() => {

        Object.defineProperty(window, 'location', {
            configurable: true,
            value: { reload: jest.fn() },
          });

        props = {
            details: {
                "volumeInfo": {
                    "id": 1,
                    "title": "Introduction to Probability",
                    "imageLinks": {
                        "smallThumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                        "thumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                    },
                    "authors": [
                        "John E. Freund"
                    ],
                    "description": "Featured topics include permutations and factorials, probabilities and odds, frequency interpretation, mathematical expectation, decision making, postulates of probability, rule of elimination, much more. Exercises with some solutions. Summary. 1973 edition.",
                    "price": 12566,
                    "quantity": 3
                }
            }
        }

        component = shallow(<AddGoogleBook {...props} />);

    })

    it('should match snapshot', () => {
        expect(component.debug).toMatchSnapshot();
    });

    it('should validate elements in search Book Inventory component', () => {

        const event = {};
        expect(component.find('.dialogTitle').length).toBe(1);
        component.find('.closeButton').at(0).simulate('click', event);
        component.find('.approveButton').at(0).simulate('click', event);
    })
})