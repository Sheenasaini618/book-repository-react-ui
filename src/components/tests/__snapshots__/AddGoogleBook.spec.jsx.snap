// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGoogleBook /> should match snapshot 1`] = `[Function]`;
